pkg/resource/internal/kv/kv.go

```diff
@@ -92,7 +92,8 @@ func (kv KV) Equal(other KV) bool {
 	}
 
 	for k, v := range kv.m {
-		if v != other.m[k] {
+		otherV, ok := other.m[k]
```

Comment:
Just drop the whole block and use `maps. Equal`

Reply:
hehe yes that actually makes sense :D